app/scripts/modules/core/src/pipeline/config/triggers/helm/HelmTriggerTemplate.tsx

```diff
@@ -0,0 +1,90 @@
+import React, { useState, useEffect } from 'react';
+
+import { AccountService } from 'core/account/AccountService';
+import { HelpField } from 'core/help/HelpField';
+import { FormField, ReactSelectInput, TextInput } from 'core/presentation';
+import { IHelmTrigger } from 'core/domain/IHelmTrigger';
+
+export interface IHelmTriggerTemplateProps {
+  trigger: IHelmTrigger;
+  isManual: boolean;
+  onHelmChanged: (changes: IHelmTriggerTemplateState) => void;
+}
+
+export interface IHelmTriggerTemplateState {
+  accounts: string[];
+  account: string;
+  chart: string;
+  version: string;
+}
+
+export const HelmTriggerTemplate = (props: IHelmTriggerTemplateProps) => {
+  const [accounts, setAccounts] = useState<IHelmTriggerTemplateState['accounts']>([]);
+  const [account, setAccount] = useState<IHelmTriggerTemplateState['account']>(props.trigger.account);
+  const [chart, setChart] = useState<IHelmTriggerTemplateState['chart']>(props.trigger.chart);
+  const [version, setVersion] = useState<IHelmTriggerTemplateState['version']>(props.trigger.version);
+
+  useEffect(() => {
+    // Fetch Helm-compatible accounts
+    AccountService.getArtifactAccounts().then(allAccounts => {
+      const accounts = allAccounts
+        .filter(a => {
+          return a.types.includes('helm/chart');
+        })
+        .map(it => it.name);
+
+      setAccounts(accounts);
+    });
+  }, []);
+
+  const stateChanged = (setter: Function, field: string, value: string) => {
+    setter(value);
```

Comment:
Do we need to store `account`, `chart`, and `version` on state, or could we read them from `props.trigger`?

Reply:
These fields are editable, and are state for that reason.

Reply:
Totally understood that these fields are editable; was more wondering if we need to duplicate the source of truth for their current value on both this component's state and whatever wrapper's state we're updating with `onHelmChanged` -- when we call `onHelmChanged` is that not updating the object that we pass back down to this component as `props.trigger`?

Reply:
Ah, I see what you mean.  I think the internal component state here is still the "real" source of truth - the `onHelmChanged` just wraps the differing details between which pipeline config fields those pieces of state map back to (it's different for manual UI and for the pipeline trigger UI), and ends up just sort of proxying the data upward.

Reply:
Ah, I see what you mean.  The actually-stored fields those pieces of state map back to are different in the manual UI and the pipeline config UI.  In order to update the parent directly, it would need to know each of those actually-stored paths down in the shared component.  The current implementations of `onHelmChanged` abstract the actually-stored paths to a level above, and are most just a thin proxy back upward.

Reply:
A, got it -- thanks for the explanation! 😄